src/Microsoft.AspNet.SignalR.Core/Infrastructure/PerformanceCounterManager.cs

```diff
@@ -438,6 +438,14 @@ public IPerformanceCounter LoadCounter(string categoryName, string counterName,
             // and when they are thrown. 
             try
             {
+                if (!PerformanceCounterCategory.Exists(categoryName))
```

Comment:
The Exists check can be very expensive, look at the Remarks section of the PerformanceCounterCategory.Exists member on msdn. It's much faster to try to open and fail than to check in advance.